src/prpy/base/robot.py

```diff
@@ -284,15 +284,33 @@ def ViolatesVelocityLimits(self, traj):
         return False
 
     def _PlanWrapper(self, planning_method, args, kw_args):
-        # Call the planner.
-        traj = planning_method(self, *args, **kw_args)
 
-        # Strip inactive DOFs from the trajectory.
+        # Call the planner.
         config_spec = self.GetActiveConfigurationSpecification()
-        openravepy.planningutils.ConvertTrajectorySpecification(traj, config_spec)
+        result = planning_method(self, *args, **kw_args)
+
+        # Define the post processing steps for the trajectory.
+        def postprocess_trajectory(traj, kw_args):
+
+            # Strip inactive DOFs from the trajectory.
+            openravepy.planningutils.ConvertTrajectorySpecification(
+                traj, config_spec
+            )
+
+            # Optionally execute the trajectory.
+            if 'execute' not in kw_args or kw_args['execute']:
+                return self.ExecuteTrajectory(traj, **kw_args)
+            else:
+                return traj
```

Comment:
Didn't realize timeout was already overloaded.  We should special-case the 0 case then.

We don't want ExecuteTrajectory to return a future in situations where people just want to write a script that does a few things, in order.  Or to minimize modification to existing code.

Unless you want to seriously bust the API and use exclusively futures everywhere, but that seems like it will massively confuse people.